### ISSUE TITLE
Add size parameter to prefer-destructuring 

### ISSUE THREAD

**Issue by andgreendigital** [eslint/eslint#11581]
**What rule do you want to change?**

prefer-destructuring

**Does this change cause the rule to produce more or fewer warnings?**

fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**

new configuration option

**Please provide some example code that this change will affect:**

```js
var foo = array[100];
```

**What does the rule currently do for this code?**

Reports an error stating "Use array destructuring"

**What will the rule do after it's changed?**

As stated in the [documentation for this page](https://github.com/eslint/eslint/blob/v5.16.0/docs/rules/prefer-destructuring.md#when-not-to-use-it) this rule doesn't make any sense for:
```js
var foo = array[100];
```
But does still make sense for:
```js
var foo = array[0];
```
If there was a configuration option for the access number then this would solve that problem! For example if set to 3 then accessing an element under `array[3]` would prefer destructuring but accessing an element greater than `array[3]` would not cause an error to prefer destructuring as in this case it wouldn't be readable.

**Are you willing to submit a pull request to implement this change?**

Unfortunately not as I am only 6 months into my career and don't have the time to learn the required skills and delve into this codebase.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal.

Could you elaborate on why you're using code like `var foo = array[100]`? It seems like there aren't a lot of cases where it would be necessary to access the hardcoded 101st element of an array.

An option like this was previously proposed in https://github.com/eslint/eslint/issues/9386, and the result of the discussion was basically that when an index is large enough that using commas is inconvenient, it's also typically large enough that one shouldn't be using an array to store it anyway.

**Comment by g-plane** [comment]
A very simple use case:

```js
const firstCLIArg = process.argv[2]
```
